numba/tests/doc_examples/test_interval_example.py

```diff
@@ -177,7 +210,23 @@ def check_equal_intervals(x, y):
         # Test .width attribute
         self.assertEqual(a.width, interval_width(a))
 
-        # Test .low and .high usage
+        # Test exceptions
+        class NotAFloat:
+            def __float__(self):
+                raise RuntimeError("I am not a float")
+
+        # TODO: the float unboxer swallows our error and converts it to a `TypeError`. This should
+        # be `RuntimeError`, but isn't the fault of the `Interval` integration were are testing.
```

Comment:
While I understand the message, I think the sentence sounds like it doesn't make any sense. Maybe also be more explicit and mention that this probably arises from calling `__float__` as part of the float unboxing via:

https://docs.python.org/3/c-api/float.html?highlight=asdouble#c.PyFloat_AsDouble

Reply:
How about this longer explanation?
```suggestion
        # TODO: This should produce a `RuntimeError`, but the `unbox` handler for `float` ignores
        # the error raised by `__float__`, which leads to a subsequent `TypeError` when passing
        # `NULL` to `PyFloat_AsDouble`.
        # This isn't the fault of the `Interval` extension were are testing in this file.
```

Reply:
(if that's good, feel free to apply both this suggestion and the previous one to get this PR moving)